src/Renderer/LayeredMaterial.js

```diff
@@ -76,7 +90,7 @@ var moveElementsArraySafe = function moveElementsArraySafe(array,index, howMany,
 };
 /* eslint-enable */
 
-const LayeredMaterial = function LayeredMaterial(id) {
+const LayeredMaterial = function LayeredMaterial(material, coordDestination, coordSource) {
```

Comment:
@peppsac , you disagree, me too I prefer a most generic thing, as you suggest.
I ask for another opinion because I think it isn't possible with this `mapping` method,  due to nonlinear projections (PM, Lambert...) This method requires several unnecessary conversions. On the other hand it is perfect in the WGS84 system
```js
mapping = {
    source: {
        origin: new THREE.Vector2(0, 0),
        size: 1,
    },
    destination: {
        origin: new THREE.Vector2(0, 0),
        size: 1,
    },
}
```